[deannapoh] iP

deannapoh opened the pull request:
**Boo**

> Today is your opportunity to build the tomorrow you want. - Ken Poirot ([source](https://www.builder.ai/blog/motivational-quotes))

Boo, your friendly assistant, will help you keep track of tasks so that you will stop ghosting your responsibilities! 👻  It's: 
- text-based
- easy to learn 
- ~FAST~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/deannapoh/ip)
2. double-click it.
3. add your tasks.
4. and let it manage your tasks for you! 😉

And best of all, it's **COMPLETELY FREE**!

Here are some of the features that Boo has:
- [x] Managing tasks
- [ ] Managing events
- [x] Managing deadlines
- [x] Reminders (COMING SOON!)

##
Here's a small snippet on how Boo works:
This is the `parseCommand(String input)` method: 

```ruby
public boolean parseCommand(String input) throws BooException {
        if (input.equalsIgnoreCase("bye")) {
            ui.printGoodbyeMessage();
            return true;
        }
        if (input.equalsIgnoreCase("list")) {
            ui.printTaskHistory(taskList.getTaskMap());
        } else if (input.toLowerCase().startsWith("mark")) {
            taskList.markAsDone(input);
        } else if (input.toLowerCase().startsWith("unmark")) {
            taskList.markAsNotDone(input);
        } else if (input.toLowerCase().startsWith("delete")) {
            taskList.deleteTask(input);
        } else if (input.toLowerCase().startsWith("find")) {
            taskList.findTask(input);
        } else {
            taskList.addTask(parseTask(input));
        }
        return false;
    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

deannapoh committed:
renamed chatbot and added greet and exit functionalities

deannapoh committed:
added echo

deannapoh committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

deannapoh committed:
Added the ability to mark tasks as done and added the ability to change the status back to not done.

deannapoh committed:
Added todo, deadline and event tasks

deannapoh committed:
Used the input/output redirection technique to semi-automate the testing of Duke.

deannapoh committed:
Taught the chatbot to deal with errors such as incorrect inputs entered by the user.

deannapoh committed:
Added support for deleting tasks from the list.

deannapoh committed:
Add feature to save and load tasks in a hard disk.

TaskList reloads when chatbot restarts.

This prevents users from keeping track of their task history.

Saving the tasks in a hard disk automatically whenever the task list
changes and loading the data from the hard disk when the chatbot starts
up allows users to better keep track of their task history.

deannapoh committed:
Add feature to save and load tasks in a hard disk

TaskList reloads when chatbot restarts.

This prevents users from keeping track of their task history.

Saving the tasks in a hard disk automatically whenever the task list
changes and loading the data from the hard disk when the chatbot starts
up allows users to better keep track of their task history.

deannapoh committed:
Merge branch 'branch-Level-7'

deannapoh committed:
Add feature to teach Chatbot to understand dates and times

User can input any type of dates and times. This makes the format
unstandardised and messy.

Teaching the Chatbot how to understand dates and times makes task list
neater and easier to read.

deannapoh committed:
Add feature to teach Chatbot to understand dates and times.

User can input any type of dates and times. This makes the format
unstandardised and messy.

Teaching the Chatbot how to understand dates and times makes task list
neater and easier to read.

deannapoh committed:
Edit code to follow OOP principles

There are some parts of the code that violate OOP principles.

Adding additional classes such as Ui and Parser can improve the code
quality.

deannapoh committed:
Organise classes into packages

All classes are under the same directory.

Organising the classes into packages allows for easier management

deannapoh committed:
Merge branch 'branch-A-Packages'

deannapoh committed:
Merge branch 'add-gradle-support'

deannapoh committed:
Add JUnit tests

Need to use manual testing to test for bugs in the code. Adding JUnit
tests help to make testing more efficient.

deannapoh committed:
Merge branch 'branch-A-JUnit'

deannapoh committed:
Add Javadoc comments

Methods have no headers describing what they do. Adding Javadoc
comments makes the method functionalities clearer and easier to
understand.

deannapoh committed:
Merge tag 'branch-A-JavaDoc'

deannapoh committed:
Edit the code such that it complies with coding standard

Code does not comply with coding standard. Thus, editing the code helps
to make the code neater.

deannapoh committed:
Edit the code such that it complies with coding standard

Code does not comply with coding standard. Thus, editing the code helps
to make the code neater.

deannapoh committed:
Edit the code such that it complies with coding standard

Code does not comply with coding standard. Thus, editing the code helps
to make the code neater.

deannapoh committed:
Edit the code such that it complies with coding standard

Code does not comply with coding standard. Thus, editing the code helps
to make the code neater.

deannapoh committed:
Add feature that allows users to find tasks

Users can only see the list of tasks they have, but they cannot find
tasks with specific keywords. This feature allows them to view their
tasks more conveniently.

deannapoh committed:
Merge branch 'branch-Level-9'

deannapoh committed:
Merge remote-tracking branch 'origin/branch-Level-8'

Fix merge issue

deannapoh committed:
Add GUI to the chatbot

Chatbot has no GUI. Adding GUI can help enhance the user experience.

deannapoh committed:
Merge tag 'branch-Level-10'

namprice227 commented on src/main/java/boo/Boo.java:
```diff
@@ -0,0 +1,67 @@
+    public String getResponse(String input) {
+        ui.printGreeting();
+        try {
+            if (input.equalsIgnoreCase("bye")) {
```
you may try switch-case to make the code easier to see

namprice227 commented on src/main/java/boo/misc/Main.java:
```diff
@@ -0,0 +1,35 @@
+
+    private Boo boo = new Boo();
+
+    public Main() throws BooException {
```
You may delete this method

namprice227 commented on src/main/java/boo/misc/Storage.java:
```diff
@@ -0,0 +1,177 @@
+                        // Convert "dd MMM yyy h:mm a" to "dd/MMM/yyy hhmm"
+                        LocalDateTime parsedFrom = LocalDateTime.parse(
+                                from, DateTimeFormatter.ofPattern("dd MMM yyyy h:mm a"));
+                        String newFrom = parsedFrom.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
```
Maybe variable naming can be improved

deannapoh replied:
Thanks for the comment! Will try to rectify it :)

namprice227 commented on src/main/java/boo/misc/Ui.java:
This file has good coding standard

namprice227 commented on src/main/java/boo/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+ * A deadline task has a description, an end date and a completion status.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
your variable naming is a bit vague

deannapoh replied:
Thanks for pointing this out! Will rectify 😃

namprice227 commented on src/main/java/boo/task/Task.java:
```diff
@@ -0,0 +1,54 @@
+        return this.description;
+    }
+
+    public boolean isDone() {
```
good that you add is for boolean var

namprice227 reviewed (commented):
Overall your code standard is good, maybe at some place variable naming cna be better to make it easier for readers to understand

deannapoh committed:
Incorporate assertions into code

There are no assertions in the code. There are only exceptions and
try-catch blocks.

Using assertions help to document important assumptions that should
hold at various points in the code. This helps in debugging.

Let's incorporate assertions in our code to make debugging easier.

deannapoh committed:
Improve code quality

There are some method and variable names that could be clearer.

Renaming these names can improve readability and understanding of the
code.

Let's provide clearer naming to improve our code quality.

deannapoh committed:
Merge pull request #1 from deannapoh/branch-A-Assertions

Incorporate assertions into code

deannapoh committed:
Resolved merge conflicts

deannapoh committed:
Resolve merge conflicts

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Add ability to re-arrange tasks in chronological order

Chatbot adds tasks based on whichever task is first input by user.

Automatically re-arranging tasks allows users to keep track of their
tasks better, improving user experience.

deannapoh committed:
Add ability to re-arrange tasks in chronological order

Chatbot adds tasks based on whichever task is first input by user.

Automatically re-arranging tasks allows users to keep track of their
tasks better, improving user experience.

deannapoh committed:
Add minor changes in naming of variables

deannapoh committed:
Add minor changes in naming of variables

deannapoh committed:
Resolve merge conflicts

deannapoh committed:
Resolve merge conflicts

deannapoh committed:
Resolve merge conflicts

deannapoh committed:
Improve GUI

deannapoh committed:
Improve GUI

deannapoh committed:
Update README.md

deannapoh committed:
Update README.md

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve Code Quality

deannapoh committed:
Merge branch 'master' of https://github.com/deannapoh/ip

deannapoh committed:
Improve Code Quality

deannapoh committed:
Improve code quality

deannapoh committed:
Improve code quality
